### PR TITLE
[WIP] dynamic-import `importHref()` callback fix when the same resource is requested multiple times

### LINKED ISSUE #3814: importHref only triggers onload and onerror if first invocation has listeners
### Description

`Polymer.Base.importHref` only triggers `onload` and `onerror` if first invocation has listeners. If another call to the same URI is made with load or error listeners these will never be called.

Imagine a scenario where two different components lazyload the same import, one of which cares for load events and one who just progressively enhance the experience. If the one which doesn't need to know whether or not the load is complete is initiated before the component that cares no load event will be triggered at all.
#### Live Demo

Example using a delayed prefetch for `paper-toast`. The button _Your toast_ must be clicked at least once before prefetching is complete to be able to show a toast using that button.
https://jsbin.com/behoyu/15/edit?html,output
#### Steps to Reproduce
1. Call `Polymer.Base.importHref('a.html', null, null, true)`
2. Wait for load to complete.
3. Call `Polymer.Base.importHref('a.html', e => console.log('onload', e), e => console.log('onerror', e), true)`

Neither `onload` nor `onerror` will be shown in the console.
#### Expected Results

Ordering of the two calls in the example should be of no consequence. The `load` or `error` listeners should be called in either case.
##### Expected Results - Example
1. Call `Polymer.Base.importHref('a.html', null, null, true)`
2. Wait for load to complete.
3. Call `Polymer.Base.importHref('a.html', e => console.log('onload', e), e => console.log('onerror', e), true)`

and
1. Call `Polymer.Base.importHref('a.html', () => {}, () => {}, true)`
2. Wait for load to complete.
3. Call `Polymer.Base.importHref('a.html', e => console.log('onload', e), e => console.log('onerror', e), true)`

should generate the same console output.
#### Actual Results

If first call is with no listeners the load and error callbacks in subsequent calls will never fire.
### Browsers Affected

All
### Versions
- Polymer: my guess is since #3638 got fixed.

### PR DESCRIPTION
**Pull request by gronke**
fixes #3814

The expected behavior of `importHref()` is that onload or onerror get called once the resource is loaded. 

When requesting a resource for the second time, we can immediately call the onload or onerror function in case it was already completed. Otherwise we attach to the same events as the previous request.
### Unit Test
- trigger load event of the second request of a resource when loading was not complete yet
- trigger load event of the second request of a resource when loading was already completed (addresses #3814)
- triggers the callback only once even when the element was requested again later

### CODE REVIEW

File: test/unit/dynamic-import.html [hunk 1]
```diff
@@ -65,6 +65,42 @@
...
+          done();
+        });
+      });
+
+      test('trigger load event of the second request of a resource when loading was already completed', function(done) {
+        var url = 'dynamic-imports/dynamic-lazy.html';
+        var imprt = Polymer.Base.importHref(url);
+
+        var onLoad = function() {
+          document.removeEventListener('load', onLoad);
```
Comment (sorvell) [inline test/unit/dynamic-import.html]: `document` here should be `imprt`

File: test/unit/dynamic-imports/dynamic-once.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
```
Comment (sorvell) [inline test/unit/dynamic-imports/dynamic-once.html]: Please remove if this file is not used anywhere.

**Comment by gronke** [comment]
@sorvell thanks for the review. all done!

**Comment by blasten** [comment]
oh this would have also fixed https://github.com/Polymer/polymer/pull/4124

**Comment by gronke** [comment]
@blasten will rebase

**Comment by gronke** [comment]
Closing in favor of #4209

**Pull request closed without merging**